### PR TITLE
hardmode update(READ CAREFULLY)

### PR DESCRIPTION
**Pull request by jiayuhan94**
Someupdates trying to see if hard mode works. the "ezmode" doesnt change to false when we select hard mode. it might be my computer problem since it always give me "Process finished with exit code 100", doesnt continue running when I was trying to debug. accept if it works. if this problem only occurs on my machine, then you could ignore this PR

### CODE REVIEW

File: src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java [hunk 1]
```diff
@@ -159,10 +159,6 @@ public void hardplace(){
         hardgenerate("computer_submarine");
         hardgenerate("computer_aircraftC");
```
Comment (edwardne) [inline src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java]: This isn't right and might be the issue

Comment (edwardne) [reply src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java]: those bottom two should be dinghy and clipper

Comment (edwardne) [reply src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java]: Also just noticed missing the AI() for aircraft carrier. The function isn't even there

**Comment by geddingj** [comment]
Alright just followed the path the code takes from setEzmode() in BattleshipModel for the current status of our program (not including this pull).

When hard mode is selected the path it takes is:
1. setEzmode() determines the difficulty
2. hardplace() is called once hard difficulty is confirmed
3. hardplace() calls hardgenerate("computer_aircraftCarrier")
4. hardgenerate() will run until the line: getShip(shipName).setLocation(start, end);

At this point the last trace statement I could get working indicated that in getShip() it returns null and the code doesn't appear to reach setLocation().

**Comment by edwardne** [comment]
getship is returning null because it is getting bad shipnames. did you see
my code review?

**Comment by geddingj** [comment]
Yeah I tried changing those to clipper and dinghy but it didn't change anything, from what I can tell the code is hanging on the first hardgenerate call for "computer_aircraftCarrier"

**Comment by edwardne** [comment]
just noticed the get ship is only checking for the players ships. doesn't
have computer_ ones

**Comment by edwardne** [comment]
I have it fixed now since I had to deal with it in my tests. Will have that
pull request in abit

**Comment by edwardne** [comment]
Is the hardgenerate() replacing all of the AIdh(), AIcl(), etc ones?

**Comment by jiayuhan94** [comment]
Yes, they should be included in my most recent or.

**Comment by jiayuhan94** [comment]
some updates on battleshipmodel test file, roughly finished battleshipmodel test coverage, have 3 methods left.
Also,
corrected some errors I found when I was writing the test

**Comment by geddingj** [comment]
Alright I think the code is in a good spot now, if nothing else the problems I was hitting earlier look to have been corrected.

Merging now.

**Pull request merged**